src/Table/Table.js

```diff
@@ -58,81 +68,149 @@ const Table = React.forwardRef(({ headers, tableData, className, compact, conden
+            selection.onSelectRow(cell.row - 1);
+            event.preventDefault();
+            return;
+        }
+
+        setInstructionsText(generateInstructionsText(cell, event));
+    };
+
+    const onClickRowHandler = (cell) => {
+        selection?.onClickRow && selection.onClickRow(cell.row - 1);
+    };
+
     const onToggleEditMode = (enable) => {
         setInstructionsText(enable ? localizedText.editModeDisable : '');
     };
 
     const onFocusCell = (cell, event) => {
-        const { row, col } = cell;
-        const key = event.which || event.keyCode;
-        const navigatedHorizontally = (key === keycode.codes.left || key === keycode.codes.right) && col > 0;
-        const navigatedVertically = (key === keycode.codes.up || key === keycode.codes.down) && row > 0;
+        setInstructionsText(generateInstructionsText(cell, event));
+    };
+
+    const onBlurTable = (event) => {
+        if (event.target === tableRef.current) {
+            setInstructionsText('');
+        }
+    };
+
+    const generateInstructionsText = (cell, event) => {
         if (gridManager.current?.editMode) {
-            setInstructionsText(localizedText.editModeDisable);
-        } else {
-            let newInstructionsText = '';
-            if (navigatedVertically) {
-                newInstructionsText += `${localizedText.row} ${row} `;
-            } else if (navigatedHorizontally) {
-                newInstructionsText += `${localizedText.column} ${col} ${headers[col]} `;
-            } else {
-                newInstructionsText += `${localizedText.arrowKeys} `;
-            }
-            if (gridManager.current?.isEditableCell(cell)) {
-                newInstructionsText += localizedText.editModeEnable;
-            }
-            setInstructionsText(newInstructionsText);
+            return localizedText.editModeDisable;
         }
+
+        let newInstructionsText = '';
+
+        switch (event.type) {
+            case 'focus':
+                if (instructionsText.length === 0) {
+                    if (keyboardNavigation !== 'row') {
+                        newInstructionsText += `${localizedText.arrowKeys} `;
+                    } else {
+                        newInstructionsText += `${localizedText.rowArrowKeys} ${localizedText.rowSelection} ${localizedText.rowClick}`;
+                    }
+                }
+                break;
+
+            case 'keydown':
+                const { row, col } = cell;
+                const key = event.which || event.keyCode;
+
+                const navigatedHorizontally = (key === keycode.codes.left || key === keycode.codes.right) && col > 0;
+                const navigatedVertically = (key === keycode.codes.up || key === keycode.codes.down) && row > 0;
+
+                if (navigatedVertically) {
+                    newInstructionsText += `${localizedText.row} ${row} `;
```

Comment:
I think this string needs to have the row number as a variable rather than concatenated. This wouldn't be grammatical in some other languages. Same for column.

Reply:
Decided to remove the row and column numbers as screen readers still give that information anyway when using the built-in table navigation keys (ctrl+option/alt+arrow on VO and JAWS). And it doesn't seem prudent to try and work in a localization system that supports variables right now.